airflow/models/dag.py

```diff
@@ -2557,6 +2569,14 @@ def bulk_write_to_db(cls, dags: Collection["DAG"], session=NEW_SESSION):
                     orm_dag.tags.append(dag_tag_orm)
                     session.add(dag_tag_orm)
 
+            orm_dag_links = orm_dag.dag_owner_links or []
+            for orm_dag_link in orm_dag_links:
+                if orm_dag_link not in dag.owner_links:
+                    session.delete(orm_dag_link)
+            for owner_name, owner_link in dag.owner_links.items():
+                dag_owner_orm = DagOwnerAttributes(dag_id=dag.dag_id, owner=owner_name, link=owner_link)
+                session.add(dag_owner_orm)
```

Comment:
Hmm this could be a bit inefficient. Can we use `IN` and bulk insert to make this at most two calls? (one for delete and one for insert)

Reply:
Can you elaborate a bit on your comment? I think I didn't fully get your intention :(
I tried to change it to use `session. bulk_save_objects ` or `session. bulk_insert_mappings`, but it's not going to work because we can't use it because it breaches the foreign key constraint, as when we try to do this, the `dag` table is not yet populated.
It gets populated in this function, but only when we call `session.flush()` here:
https://github.com/apache/airflow/blob/main/airflow/models/dag.py#L2623

AFAIK, running `session.add()` multiple times is adding a transaction, but it's "lazy" because nothing happens until we run `flush()`. BTW, this implementation is using the same flow as we use to update the tags, here:
https://github.com/apache/airflow/blob/main/airflow/models/dag.py#L2547
So this is why I used it 😅 

Can you please explain a bit more or provide an example of the intention of this comment, so maybe I'll understand it more clearly and be able to add this enhancement?